src/utils/purge.js

```diff
@@ -24,3 +42,19 @@ module.exports = url => {
       throw e;
     });
 };
+
+/**
+ * Generic entrypoint for authenticated/unauthenticated purge reuqests.
+ * @param {string} url
+ * @returns {Promise<Object>} An object containing the request data
+ * @throws {Error} Throws an error if the request was not successful
+ */
+const purge = function (url) {
```

Comment:
Note: I created this dispatch function here rather than putting the conditionals in a single function (and using them to set the `fetch` parameters). I think that will give us a little more flexibility if we need to modify error handling for one method down the road, or if we add a third method and end up with a `case` statement for dispatch.